webapp/src/ts/modules/messages/messages-content.component.ts

```diff
@@ -0,0 +1,322 @@
+import { SendMessageComponent } from '@mm-modals/send-message/send-message.component';
+
+/**
+*  In this context the URL parameter "id", can be:
+*    - the _id of the contact who is sending these messages
+*    - the phone number if it couldn't be resolved into a contact
+*    - the _id of the data_record if there is no discernible phone number
+*  This is determined by the URL parameter "type": 'contact', 'phone' or 'unknown' respectively.
+*/
+@Component({
+  selector: 'messages-content',
+  templateUrl: './messages-content.component.html'
+})
+export class MessagesContentComponent implements OnInit, OnDestroy, AfterViewInit {
+  private userCtx;
+  private globalActions: GlobalActions;
+  messagesActions: MessagesActions;
+  loadingContent;
+  loadingMoreContent = false;
+  selectedConversation;
+  firstUnread;
+  send = { message: '' };
+  allLoaded = false;
+  urlParameters = { type: '', id: '' };
+  subscriptions: Subscription = new Subscription();
+  textAreaFocused = false;
+
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    private route: ActivatedRoute,
+    private store: Store,
+    private changesService: ChangesService,
+    private messageContactService: MessageContactService,
+    private sessionService: SessionService,
+    private lineageModelGeneratorService: LineageModelGeneratorService,
+    private markReadService: MarkReadService,
+    private sendMessageService: SendMessageService,
+    private modalService: ModalService,
+  ) {
+    const selectorsSubscription = combineLatest(
+      this.store.pipe(select(Selectors.getSelectedConversation)),
+      this.store.pipe(select(Selectors.getLoadingContent)),
+    ).subscribe(([selectedConversation, loadingContent]) => {
+      this.selectedConversation = selectedConversation;
+      this.loadingContent = loadingContent;
+    });
+    this.subscriptions.add(selectorsSubscription);
+
+    this.globalActions = new GlobalActions(store);
+    this.messagesActions = new MessagesActions(store, this.globalActions);
+  }
+
+  ngOnInit(): void {
+    this.userCtx = this.sessionService.userCtx();
+    this.watchForChanges();
+  }
+
+  ngAfterViewInit(): void {
+    const routeSubscription = this.route.params.subscribe(params => {
+      const [type, id] = params.type_id ? params.type_id.split(':') : [];
+      this.urlParameters.type = type;
+      this.urlParameters.id = id;
+      this.send.message = '';
+      this.selectContact(this.urlParameters.id, this.urlParameters.type);
+      this.changeDetectorRef.detectChanges();
+    });
+    this.subscriptions.add(routeSubscription);
+
+    // Ensuring that any Bootstrap tooltip is removed when loading new conversation.
+    $('.tooltip').remove();
+    $('body')
+      .on('focus', '#message-footer textarea', () => {
+        this.textAreaFocused = true;
+      })
+      .on('blur', '#message-footer textarea', () => {
+        // Setting timeout to give change for clicking "add recipient" before hiding section.
```

Comment:
It's hiding when the textarea looses focus, then it doesn't give time to click on the buttons. I think it makes more sense that we hide if the focus is lost from the container wrapping the textarea and the buttons instead of just textarea. What do you think about this idea?

Reply:
You can give it a try, most important is to have the same behavior as before.

Reply:
I think it's fixed now. I had to refactored a bit.